lib/Sema/CSSimplify.cpp

```diff
@@ -3781,16 +3798,59 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyMemberConstraint(
 
     // Instead of using subscript operator spelled out `subscript` directly.
     if (member.getBaseName() == getTokenText(tok::kw_subscript)) {
-      auto result = simplifyMemberConstraint(
-          kind, baseTy, DeclBaseName::createSubscript(), memberTy, useDC,
-          functionRefKind, {}, flags, locatorB);
-
+      auto result =
+          solveWithNewBaseOrName(baseTy, DeclBaseName::createSubscript());
       // Looks like it was indeed meant to be a subscript operator.
       if (result == SolutionKind::Solved)
         return recordFix(UseSubscriptOperator::create(*this, locator))
                    ? SolutionKind::Error
                    : SolutionKind::Solved;
     }
+
+    // FIXME(diagnostics): This is more of a hack than anything.
+    // Let's not try to suggest that there is no member related to an
+    // obscure underscored type, the real problem would be somewhere
+    // else. This helps to diagnose pattern matching cases.
+    {
+      if (auto *metatype = baseTy->getAs<MetatypeType>()) {
+        auto instanceTy = metatype->getInstanceType();
+        if (auto *NTD = instanceTy->getAnyNominal()) {
+          if (NTD->getName() == getASTContext().Id_OptionalNilComparisonType)
```

Comment:
You can use .is(...) instead of == getIdentifier(...)

Reply:
Fixed!

Reply:
Right, that’s what I thought too, I also looked at other uses of Id_* cached identifiers